examples/nidcpower_nivisa_dmm_measurement/measurement.py

```diff
@@ -0,0 +1,242 @@
+        pin_or_relay_names=[input_pin, output_pin],
+        timeout=RESERVATION_TIMEOUT_IN_SECONDS,
+    ) as reservation:
+        grpc_device_channel = get_grpc_device_channel(
+            measurement_service, nidcpower, service_options
+        )
+        source_session_info = _get_session_info_from_pin(reservation.session_info, input_pin)
+        measure_session_info = _get_session_info_from_pin(reservation.session_info, output_pin)
+        resource_manager = create_visa_resource_manager(service_options.use_simulation)
+        # Creates NI-DCPower and NI-VISA DMM sessions.
+        with create_session(
+            source_session_info, grpc_device_channel
+        ) as source_session, create_visa_session(
+            resource_manager, measure_session_info.resource_name
+        ) as measure_session:
+            pending_cancellation = False
+            _add_cancel_callback(source_session, measurement_service, pending_cancellation)
+
+            assert isinstance(measure_session, pyvisa.resources.MessageBasedResource)
+
+            log_instrument_id(measure_session)
+
+            # When this measurement is called from outside of TestStand (session_exists == False),
+            # reset the instrument to a known state. In TestStand, ProcessSetup resets the
+            # instrument.
+            if not measure_session_info.session_exists:
+                reset_instrument(measure_session)
+            # Source input pin
+            channels = source_session.channels[source_session_info.channel_list]
+
+            channels.source_mode = nidcpower.SourceMode.SINGLE_POINT
+            channels.output_function = nidcpower.OutputFunction.DC_VOLTAGE
+            channels.current_limit = current_limit
+            channels.voltage_level_range = voltage_level_range
+            channels.current_limit_range = current_limit_range
+            channels.source_delay = hightime.timedelta(seconds=source_delay)
+            channels.voltage_level = voltage_level
+            with channels.initiate():
+                _wait_for_source_complete_event(measurement_service, channels, pending_cancellation)
+
+            # Configure and measure output pin using NI-VISA DMM
+            function_enum = FUNCTION_TO_VALUE[measurement_type]
+            resolution_value = RESOLUTION_DIGITS_TO_VALUE[str(resolution_digits)]
+            measure_session.write("CONF:%s %.g,%.g" % (function_enum, range, resolution_value))
+            check_instrument_error(measure_session)
+
+            response = measure_session.query("READ?")
+            check_instrument_error(measure_session)
+            measured_value = float(response)
+
+            source_session = None  # Don't abort after this point
+
+    logging.info("Completed measurement: measured_value=%g", measured_value)
+    return (measured_value,)
+
+
+def _get_session_info_from_pin(session_info, pin_name):
+    session_index = get_session_and_channel_for_pin(session_info, pin_name)[0]
```

Comment:
will this list be empty? should be checked?

Reply:
`session_info` will only be empty if the existing code gets edited.
Hence, Not handling empty lists should be okay in this context.